src/primitives/transaction.h

```diff
@@ -43,7 +44,10 @@ class COutPoint
 
     friend bool operator<(const COutPoint& a, const COutPoint& b)
     {
-        return std::tie(a.hash, a.n) < std::tie(b.hash, b.n);
+        DataStream ser_a, ser_b;
+        ser_a << a;
+        ser_b << b;
+        return Span{ser_a} < Span{ser_b};
```

Comment:
7a27533332d14cf98f7e165f464142132e9f9d85: if you're going to change this here instead of adding a custom sort for BIP352, then you should add a note that it's critical for BIP352.

However I'm not sure if we should sort this way by default. If I wanted to list transaction outputs, I'd want to sorted by their integer version.

That said, I don't know if we currently expose any RPC where this sort order is exposed.

Reply:
From what I could tell, we don’t expose this anywhere. I’m in favor of making this the default since we only ever really talking about outpoints in terms of their serialised encoding. Gonna give it some more thought, but I’m considering opening this commit as its own PR. It seems preferable to me to have one way of sorting outpoints vs creating a special case for silent payments, especially when there doesn’t seem to be any other “sort outpoints” use case (e.g. none of the tests failed when I made this change).

Reply:
But still need to verify re: displayed in RPCs. At that point tho, we could just sort by `vout` and ignore the txid in the sort

Reply:
Separate PR sounds good to get more eyes on it. Meanwhile this is fine a temporary fix for this PR.

Reply:
Closed #30046 :smile: